src/gensxg.c

```diff
@@ -449,23 +458,34 @@ static void print_integrity_hash(const sxg_encoded_response_t* encoded) {
   }
 }
 
-void print_sxg(const sxg_signer_list_t* signers,
+void write_sxg(const sxg_signer_list_t* signers,
                const char* url,
-               sxg_encoded_response_t* encoded) {
+               sxg_encoded_response_t* encoded,
+               const char* output) {
   sxg_buffer_t result = sxg_empty_buffer();
   if (!sxg_generate(url, signers, encoded, &result)) {
     fputs("Failed to generate SXG.\n", stderr);
     exit(EXIT_FAILURE);
   }
-    
-  if (freopen(NULL, "wb", stdout) == NULL) {
-    perror("reopen stdout");
-    exit(EXIT_FAILURE);
+
+  FILE* out;
+  if (strcmp(output, "-") == 0) {
+    out = freopen(NULL, "wb", stdout);
```

Comment:
Is this actually required or would `out = stdout` suffice?  If it is required, presumably to open as binary, please add a comment.

Reply:
In linux environment, `stdout` is binary mode by default, so it is not actually required.
But other environment may be different, so priority was given to clarity.